### DISCUSSION TITLE
Extra Keyword Arguments Given using uart.init

### DISCUSSION THREAD

**Discussion by HopWorks** [micropython/micropython#11535] [RP2040 / Pico] [answered]
Hello,

I am using Micropython on a PI Pico W. with Thonny 4.0.2 and I am trying to use uart.init to set parameters but I keep getting an error.
USING...
```
>>> sys.implementation
(name='micropython', version=(1, 20, 0), _machine='Raspberry Pi Pico W with RP2040', _mpy=4358)
>>> sys.platform
'rp2'
```
`TypeError: extra keyword arguments given`

```
uart = UART(0, tx=Pin(12), rx=Pin(13))
uart.init(115200, 8, None, 0, timeout=10, timeout_char=10, read_buf_len=512)
```
And I also tried...
```
uart = UART(0, tx=Pin(12), rx=Pin(13))
uart.init(baudrate=115200, bits=8, parity=None, stop=0, timeout=10, timeout_char=10, read_buf_len=512)
```
with the same results. 

I hope I fully read [https://docs.micropython.org/en/latest/library/pyb.UART.html](url) correctly. 
I googled this for an hour and cannot find where I went wrong. 

Thank you for your time!!

**Comment by jimmo** [comment, marked as answer]
> `uart.init(115200, 8, None, 0, timeout=10, timeout_char=10, read_buf_len=512)`

There is no `read_buf_len` on `machine.UART`. You want `rxbuf` I think.

> I hope I fully read [https://docs.micropython.org/en/latest/library/pyb.UART.html](https://github.com/orgs/micropython/discussions/url) correctly.

That's the docs for the (legacy) `pyb.UART` class that just exists on stm32/pyboard. You want the docs for `machine.UART`. 
https://docs.micropython.org/en/latest/library/machine.UART.html

**Reply by HopWorks** [reply]
That is interesting. I thought they were all the same thing. Thank you for that clarification!!
It is too bad the error thrown could not have said there was an unknown named parameter instead of 'extra keyword arguments given'. That might have made a difference in my search criteria, but I probably would have ended up at the same dead-end. 
This also helped me to figure out why readln() didn't work either.
Thank you sir for your reply!!
